src/ibmca_ec.c

```diff
@@ -0,0 +1,918 @@
+/*
+ * Copyright [2005-2018] International Business Machines Corp.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+#include <stdlib.h>
+#include "ibmca.h"
+#include "e_ibmca_err.h"
+
+#ifndef OPENSSL_NO_EC
+
+ #ifdef OLDER_OPENSSL
```

Comment:
I think it would be better if we could do just one ifdef OLDER_OPENSSL and put all the functions that are only for old openssl version, and put all the functions that are new openssl version only in the else statement. And the functions which are shared between both versions put before (or after) this ifdef..else...endif
This would make the maintenance of the code easier. 
You can use ibmca_rsa.c as an example, it follows this structure.

Reply:
done